packages/editor/src/components/url-input/index.js

```diff
@@ -22,13 +22,23 @@ import { addQueryArgs } from '@wordpress/url';
 // as being considered from the input.
 const stopEventPropagation = ( event ) => event.stopPropagation();
 
+function getAnchorRect( anchor ) {
+	/*
+	 * The default getAnchorRect() gets the parent node to calculate the popover
+	 * position. As the popover with the list of suggestions is now nested within
+	 * another popover, we need to get the gran parent node.
+	 */
+	return anchor.parentNode.parentNode.getBoundingClientRect();
```

Comment:
I understand the logic, but I'd prefer if we find another way as this seems fragile. It used to work properly before without it (even with two popovers) so curious to know why it regressed.

Reply:
Yeah I agree it's fragile, that's why I've mentioned that maybe the popover component should handle these nesting cases internally. It happened to work before just by coincidence 🙂 If you look at the screenshot posted above, the suggestions list is perfectly centered to the first popover _highlighted area_. The ellipsis button is now outside of the first popover and doesn't contribute to determine its width, so the position calculation of the _second_ popover takes into consideration only the highlighted area. If you move the ellipsis button back inside the first popover, then the position will work correctly (but just by coincidence).